api/types/lock_test.go

```diff
@@ -112,3 +114,22 @@ func TestLockTargetMatch(t *testing.T) {
 	// Test that a lock with ServerID field set matches any target with ServerID.
 	require.True(t, targetServerID.Match(lockServerID))
 }
+
+// TestLockTargetIsEmpty checks that the implementation of [LockTarget.IsEmpty]
+// is correct by filling one field at a time and expecting IsEmpty to return
+// false. Only the public fields that don't start with `XXX_` are checked (as
+// those are gogoproto-internal fields).
+func TestLockTargetIsEmpty(t *testing.T) {
+	require.True(t, (LockTarget{}).IsEmpty())
+
+	for i, field := range reflect.VisibleFields(reflect.TypeOf(LockTarget{})) {
+		if strings.HasPrefix(field.Name, "XXX_") {
+			continue
+		}
+
+		var lt LockTarget
+		// if we add non-string fields to LockTarget we need a type switch here
+		reflect.ValueOf(&lt).Elem().Field(i).SetString("nonempty")
+		require.False(t, lt.IsEmpty(), "field name: %v", field.Name)
+	}
+}
```

Comment:
I love the idea of using reflection-based tests to ensure non-reflecting cmp is done correctly, this is fantastic!